MediaWikiKit/MediaWikiKit/MWKArticle.h

```diff
@@ -99,6 +99,25 @@ static const NSInteger kMWKArticleSectionNone = -1;
 - (MWKImage*)importImageURL:(NSString*)url sectionId:(int)sectionId;
 
 /**
+ *  Add the given URL to the receiver's image list, as well as to the corresponding section's image list.
+ *
+ *  @param sourceURL The URL string to add.
+ *  @param sectionId The section whose image list the URL should also be added to.
+ *  @param skipIfPresent @c YES if the image should only be added if the list doesn't contain it.
+ */
+- (void)updateImageListsWithSourceURL:(NSString*)sourceURL inSection:(int)sectionId skipIfPresent:(BOOL)skipIfPresent;
```

Comment:
yeah, i wrote a `HAX` comment about it, but the data layer is very error-prone when it comes to duplicating or accumulating data.  i'll change it to say `add` to be more explicit about its typical usage, but anytime you're in `MWK` land you need to be careful of this sort of thing 😰